helpers.py

```diff
@@ -45,26 +45,19 @@ def get_nested(meter_data, path):
     return value
 
 
-def get_default_template_config(config: dict, template_number: int, name: str, defaultvalue: any = None) -> any:
-    '''check if config value exist in TEMPLATE section, otherwise return defaultvalue'''
-    if name in config[f"TEMPLATE{template_number}"]:
-        return config[f"TEMPLATE{template_number}"][name]
-
-    return defaultvalue
-
-
-def try_get_value(value: str, expected_type: [str, int, float, bool],
-                  new_value: [None, str, int, float, bool]) -> [None, str, int, float, bool]:
+def parse_to_expected_type(value: str, expected_type: [str, int, float, bool],
+                           default: [None, str, int, float, bool]) -> [None, str, int, float, bool]:
     ''' Try to convert value to expected_type, otherwise return new_value'''
     try:
-        if expected_type == int:
-            return int(value)
-        if expected_type == float:
-            return float(value)
-        if expected_type == bool:
-            return is_true(value)
-    except Exception:
-        return new_value
+        conversion_functions = {
+            str: str,
+            int: int,
+            float: float,
+            bool: is_true
+        }
+        return conversion_functions[expected_type](value)
+    except (ValueError, TypeError, KeyError):
+        return default
 
     return value  # fallback
```

Comment:
This fallback will probably never happen. Only in case of an exception which is not caught here. So this might be academic: Shouldn't we better return default to fullfill the expecation ("contract") to return a value of expected_type or default?

Reply:
BTW. I couldn't even think of a test case for this fallback ;-)